src/test/java/com/google/sps/AbuseDetectionTest.java

```diff
@@ -0,0 +1,225 @@
+    instance.addRequest(LocalTime.parse("23:59:59.360"));
+    instance.addRequest(LocalTime.parse("23:59:59.590"));
+    instance.addRequest(LocalTime.parse("23:59:59.650"));
+    instance.addRequest(LocalTime.parse("23:59:59.740"));
+    instance.addRequest(LocalTime.parse("23:59:59.930"));
+    instance.addRequest(LocalTime.parse("23:59:59.980"));
+  }
+
+
+  public boolean add20Requests(AbuseDetection instance){
+    instance.addRequest(LocalTime.parse("18:19:21.000"));
+    instance.addRequest(LocalTime.parse("18:19:21.015"));
+    instance.addRequest(LocalTime.parse("18:19:21.026"));
+    instance.addRequest(LocalTime.parse("18:19:21.036"));
+    instance.addRequest(LocalTime.parse("18:19:21.047"));
+    instance.addRequest(LocalTime.parse("18:19:21.070"));
+    instance.addRequest(LocalTime.parse("18:19:21.090"));
+    instance.addRequest(LocalTime.parse("18:19:21.124"));
+    instance.addRequest(LocalTime.parse("18:19:21.145"));
+    instance.addRequest(LocalTime.parse("18:19:21.231"));
+    instance.addRequest(LocalTime.parse("18:19:21.271"));
+    instance.addRequest(LocalTime.parse("18:19:21.344"));
+    instance.addRequest(LocalTime.parse("18:19:21.421"));
+    instance.addRequest(LocalTime.parse("18:19:21.560"));
+    instance.addRequest(LocalTime.parse("18:19:21.590"));
+    instance.addRequest(LocalTime.parse("18:19:21.612"));
+    instance.addRequest(LocalTime.parse("18:19:21.642"));
+    instance.addRequest(LocalTime.parse("18:19:21.723"));
+    instance.addRequest(LocalTime.parse("18:19:21.800"));
+    boolean value = instance.addRequest(LocalTime.parse("18:19:21.900"));
+    return value;
+  }
+
+  /*
+  * 1 request is added. The timestamp is: 18:19:20.010.
+  */ 
+  @Test
+  public void addingOneTimeRequest() {
+    Duration currentDur = Duration.ofSeconds(1);
+    AbuseDetection instance = new AbuseDetection(currentDur, 10);
+    LocalTime inputTime = LocalTime.parse("18:19:20.010");
+    boolean value = instance.addRequest(inputTime);
+    // 1 request since 18:19:20.010, since it's less than 10 the 
+    // request adds and thus returns true
+    Assert.assertEquals(true, value);
+  }
+
+  /*
+  * 10 Requests are added. First timestamp: 18:19:20.010. The
+  * Last Timestamp 18:19:20.866. 
+  */
+  @Test
+  public void addingMultipleRequestsGetFalse() {
+    Duration currentDur = Duration.ofSeconds(1);
+    AbuseDetection instance = new AbuseDetection(currentDur, 10);
+    add10Requests(instance);
+    LocalTime inputTime = LocalTime.parse("18:19:20.901");
+    boolean value = instance.addRequest(inputTime);
+    // 10 requests were added since 18:19:20.010, thus the 11th within 
+    // 1 second at 18:19:20.901 fails.
+    Assert.assertEquals(false, value);
+  }
+
+  /*
+  * 10 Requests are added. First timestamp: 18:19:20.010. The
+  * Last Timestamp 18:19:20.866. 
+  */
+  @Test
+  public void addingMultipleRequestsGetTrue() {
+    Duration currentDur = Duration.ofSeconds(1);
```

Comment:
nit: Duration can be passed in line (new AbuseDetection(Duration.ofSeconds(1), 10))

Reply:
Did that.